content/en/docs/ibm/install-kubeflow.md

```diff
@@ -198,44 +210,100 @@ Run the following commands to set up and deploy Kubeflow.
+
+You will need to install [docker](https://www.docker.com/) and [kpt](https://googlecontainertools.github.io/kpt/) to update 2 configuration files.
+
+1. Create a new OAuth app in GitHub. Please keep the generated `client_id` and `client_secret`. Before exposing your kubeflow to internete, you can use following local URLs work via port-forward only:
+    * Homepage URL: `http://localhost:8080/`
+    * Authorization callback URL: `http://localhost:8080/dex/callback`
+1. Setup environment variables:
     ```
-    # The following command is optional, to make kfctl binary easier to use.
-    export PATH=$PATH:<path to where kfctl was unpacked>
-
-    # Set KF_NAME to the name of your Kubeflow deployment. This also becomes the
-    # name of the directory containing your configuration.
-    # For example, your deployment name can be 'my-kubeflow' or 'kf-test'.
     export KF_NAME=<your choice of name for the Kubeflow deployment>
 
     # Set the path to the base directory where you want to store one or more 
     # Kubeflow deployments. For example, /opt/.
-    # Then set the Kubeflow application directory for this deployment.
     export BASE_DIR=<path to a base directory>
-    export KF_DIR=${BASE_DIR}/${KF_NAME}
-
-    # Set the configuration file to use, such as the file specified below:
-    export CONFIG_URI="https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_ibm.yaml"
 
+    # Then set the Kubeflow application directory for this deployment.
+    export KF_DIR=${BASE_DIR}/${KF_NAME}
+    ```
+1. Setup configuration files:
+    ```
+    export CONFIG_URI="https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_ibm_dex.yaml"
     # Generate and deploy Kubeflow:
     mkdir -p ${KF_DIR}
     cd ${KF_DIR}
+    kfctl build -V -f ${CONFIG_URI}
+    ```
+1. Fill in an application secret for OIDC auth service in downloaded YAML file `${KF_DIR}/kustomize/oidc-authservice/oidc-authservice-cm.yaml`. Notice this application secret can be a random string but must match that in `${KF_DIR}/kustomize/dex-auth/dex-auth-cm.yaml`.
+    ```
+    curl -L -o kustomize/oidc-authservice/oidc-authservice-cm.yaml https://raw.githubusercontent.com/kubeflow/manifests/master/stacks/ibm/fns/oidc-authservice-cm.yaml
+    # update the configurations in the oidc-authservice-cm.yaml
+    ```
+1. Fill in saved GitHub OAuth app configurations like `client_id`,  `client_secret` and `application_secret` in downloaded YAML file `${KF_DIR}/kustomize/dex-auth/dex-auth-cm.yaml`:
+    ```
+    curl -L -o kustomize/dex-auth/dex-auth-cm.yaml https://raw.githubusercontent.com/kubeflow/manifests/master/stacks/ibm/fns/dex-auth-cm.yaml
+    # update the configurations in the dex-auth-cm.yaml
+    ```
+1. Transform configuration files by `kpt`:
+    ```
+    kpt fn run kustomize/oidc-authservice
```

Comment:
any reason why kpt is better here? typically discourage custom/new tools unless we have a strong reason?

Reply:
This is under discussion in https://github.com/kubeflow/kfctl/pull/364

The biggest benefit is avoid touching generated `kustomization.yaml` by hand. the preferred path of customize a kubeflow deployment is downloading configmap, tweak it then merge it via a `kpt fn`.

If @adrian555 agrees, I can fall back by documenting manually adding required `configmapGenerator` and highlight not committing to git repo, so that it won't introduce `kpt`.

Reply:
@adrian555 do you mind if I alter those the `kpt fn` path with:

1. create two configmaps locally with credentials
1. `kubectl apply -f dex-cm.yaml`
1. `kubectl apply -f oidc-service-cm.yaml`
1. `kubectl rollout restart deploy/dex -n auth`
1. `kubectl rollout restart sts/oidc-service -n istio-system`

The essential complexity of such effort is about secret management of open source components, there's no easy solution if keeping those credentials in configmaps.

Reply:
Yeah, let's go with the manual approach for now.